docs/concepts.md

```diff
@@ -35,91 +46,102 @@ When there is late arriving data (data intended for 9:00 arriving >1 hr late at
+| Query Latency | Lower (raw columnar performance) | Higher (merge columnar + row based delta) |
 
-## Copy On Write
 
-As mentioned above, each commit on Copy On Write storage, produces new versions of files. In other words, we implicitly compact every
-commit, such that only columnar data exists. As a result, the write amplification (number of bytes written for 1 byte of incoming data)
- is much higher, where read amplification is close to zero. This is a much desired property for a system like Hadoop, which is predominantly read-heavy.
+## Copy On Write Storage
+
+File slices in Copy-On-Write storage only contain the base/columnar file and each commit produces new versions of base files. 
+In other words, we implicitly compact on every commit, such that only columnar data exists. As a result, the write amplification 
+(number of bytes written for 1 byte of incoming data) is much higher, where read amplification is zero. 
+This is a much desired property for analytical workloads, which is predominantly read-heavy.
 
 Following illustrates how this works conceptually, when  data written into copy-on-write storage  and two queries running on top of it.
 
 
 {% include image.html file="hudi_cow.png" alt="hudi_cow.png" %}
 
 
-As data gets written, updates to existing file ids, produce a new version for that file id stamped with the commit and
-inserts allocate a new file id and write its first version for that file id. These file versions and their commits are color coded above.
+As data gets written, updates to existing file groups, produce a new slice for that file group stamped with the commit and
+inserts allocate a new file group and write its first version for that file id. These file slices and their commit instant times are color coded above.
 Normal SQL queries running against such dataset (eg: `select count(*)` counting the total records in that partition), first checks the timeline for latest commit
-and filters all but latest versions of each file id. As you can see, an old query does not see the current inflight commit's files colored in pink,
+and filters all but latest file slices of each file group. As you can see, an old query does not see the current inflight commit's files colored in pink,
 but a new query starting after the commit picks up the new data. Thus queries are immune to any write failures/partial writes and only run on committed data.
 
-The intention of copy on write storage, is to fundamentally improve how datasets are managed today on Hadoop through
+The intention of copy on write storage, is to fundamentally improve how datasets are managed today through
 
   - First class support for atomically updating data at file-level, instead of rewriting whole tables/partitions
   - Ability to incremental consume changes, as opposed to wasteful scans or fumbling with heuristical approaches
   - Tight control file sizes to keep query performance excellent (small files hurt query performance considerably).
```

Comment:
s/Tight/Tightly

Reply:
I think this is ok. its used as a subject in sentences above as well